js/functions.js

```diff
@@ -0,0 +1,68 @@
+/*
+Иван, добрый день! Подскажите, такой вариант первой функции будет верный? :
+const getLengthCheck = (string, maxLength) => { (string.length <= maxLength) ? true : false};
+Проверял в консоли Chrome и всегда возвращалось "undefined".
+Работало только так: const getLengthCheck = (string, maxLength) => { (string.length <= maxLength) ? console.log(true) : console.log(false)};
+*/
+//Функция для проверки длины строки:
+const getLengthCheck = (string, maxLength) => {
+  if (string.length <= maxLength) {
+    return true;
+  }
+  return false;
+}
+
+//Функця для проверки, является ли строка палинтромом + случай, когда в строке встречаются пробелы:
+const getpalindromCheck = (string) => {
+  let newString = string.toLowerCase();
+    newString = newString.replaceAll(' ', '');
+  let mirrorString = '';
+    for (i = newString.length - 1; i >= 0; i--) {
+      mirrorString += newString[i];
+    }
+  if (newString === mirrorString) {
+   return true;
+  }
+  return false;
+}
+
+/*
+Функция, которая принимает строку, извлекает содержащиеся в ней цифры от 0 до 9 и возвращает их
+в виде целого положительного числа. Если в строке нет ни одной цифры, функция должна вернуть NaN
++ случай, когда вместо строки приходит число:
+*/
+const getNumber = (data) => {
+  const string = String(data);
+  let newString = string.replace(/[^0-9]/g, '');
+  if (newString.length == 0) {
```

Comment:
Сравнивать почти всегда (лучше всегда) только через `===` строгое сравнение (это обязательный критерий), если нужно заранее поготавливать типы к одинаковым и затем сравнивать, никогда не стоит надеяться на интерпретатор.

Исключения только при использованиее `!` или других булевых операторов, когда мы в целом проверяем на наличие, чтобы сразу проверить на `0`, на пустую строку `''`, на `null`, на `undefined`, на `false` на один раз, например `if (!someExample)`  будет выполняться тогда когда someExample является одним из только что перечисленных "пустых" значений
```suggestion
  if (newString.length === 0) {
```